src/hooks/useEvidences.js

```diff
@@ -0,0 +1,74 @@
+import isIPFS from 'is-ipfs'
+import { useEffect, useState } from 'react'
+import { ipfsGet } from '../lib/ipfs-utils'
+import dayjs from '../lib/dayjs'
+import { ERROR_TYPES } from '../types/evidences-status-types'
+
+export default function useEvidences(evidences) {
+  const [evidenceProcessed, setEvidenceProcessed] = useState([])
+
+  useEffect(() => {
+    let cancelled = false
+    const appendEvidence = evidence => {
+      if (cancelled) {
+        return
+      }
+
+      setEvidenceProcessed(prev =>
+        [
+          ...prev,
+          evidence,
+        ].sort(({ createdAt: dateLeft }, { createdAt: dateRight }) =>
+          dayjs(dateLeft).isAfter(dayjs(dateRight))
+            ? 1
+            : dayjs(dateLeft).isSame(dayjs(dateRight))
+            ? 0
+            : -1
+        )
+      )
+    }
+
+    evidences.forEach(async evidence => {
+      const { data: evidenceData, submitter, createdAt } = evidence
+
+      const ipfsHash = evidenceData.replace(/^ipfs:/, '')
+
+      // check if the metadata inside the evidence is a cid
+      // if the ipfs tag is not present return the metadata as plain string
+      if (isIPFS.multihash(ipfsHash) || isIPFS.cid(ipfsHash)) {
+        const { data, error } = await ipfsGet(ipfsHash)
+        if (error) {
```

Comment:
As we handle generic errors later, it may read better if we move this to be after the `if (data)`